controllers/agentaction_controller.go

```diff
@@ -345,6 +350,41 @@ func (r *AgentActionReconciler) createWorkdirSecret(ctx context.Context, log log
 	return secret, nil
 }
 
+// creates a secret for the porter configuration directory
+func (r *AgentActionReconciler) getImagePullSecret(ctx context.Context, log logr.Logger, action *porterv1.AgentAction, agentCfg porterv1.AgentConfigSpec) (*corev1.Secret, error) {
+
+	installationSvcAccountName := "default"
+	if agentCfg.InstallationServiceAccount != "" {
+		installationSvcAccountName = agentCfg.InstallationServiceAccount
+	}
+
+	log.V(Log4Debug).Info("checking service accounts for image pull secrets", "installation_service_account", installationSvcAccountName, "action_name", action.Name, "action_namespace", action.Namespace)
+
+	var instSvcAccount corev1.ServiceAccount
+	if err := r.Get(ctx, types.NamespacedName{Namespace: action.Namespace, Name: installationSvcAccountName}, &instSvcAccount); err != nil {
+		return nil, errors.Wrap(err, "error checking for a service account")
+	}
+	log.V(Log4Debug).Info("found service account for image pull secrets", "name", instSvcAccount.Name, "number_image_pull_secrets", len(instSvcAccount.ImagePullSecrets))
+	var err error
+	var imgPullSec corev1.Secret
+	for _, secObjRef := range instSvcAccount.ImagePullSecrets {
+		log.V(Log4Debug).Info("looking for service account image pull secrets", "sa_namespace", instSvcAccount.Namespace, "sa_name", instSvcAccount.Name, "secret_name", secObjRef.Name)
+		if err := r.Get(ctx, types.NamespacedName{Namespace: action.Namespace, Name: secObjRef.Name}, &imgPullSec); err != nil {
+			log.V(Log4Debug).Info("no image pull secret found for service account", "sa_namespace", instSvcAccount.Namespace, "sa_name", imgPullSec.Name, "secret_name", imgPullSec.Name)
+		}
+		log.V(Log4Debug).Info("found image pull secret found for service account", "sa_namespace", instSvcAccount.Namespace, "sa_name", imgPullSec.Name, "secret_name", imgPullSec.Name)
+		//TODO:(bdegeeter) grab the first dockerconfig json for now. Need to address multiple image pull secrets in a single service account
```

Comment:
Can you make this into an issue in the repo so that it's easier to remember/track that more work is needed?